### DISCUSSION TITLE
Option to install all dev packages for vendors

### DISCUSSION THREAD

**Discussion by designermonkey** [composer/composer#10369] [General]
I am wondering if there could be a discussion around allowing a user of composer to install all the dev dependencies and autoloads of all or selected vendor packages?

I use composer to manage a project that has components separated into smaller sub-directories, and instead of loading vendor packages for each of those components (soo much duplication and overhead) I would like to be able to do that with a root level composer file, but I have found that I can't load any tests or test dependencies easily.

In the root file I've had to classmap the component tests which has found the files, however none of the components dev-dependencies are loaded.

So, would it be difficult to do this?

Full disclosure, I'm using `composer/installers` alongside `oomphinc/composer-installers-extender` to have components in a different directory.

**Comment by Seldaek** [comment]
That sounds frankly like a horrible change to introduce :) It's such a core assumption that only require is taken into account for dependencies, adding require-dev conditionally would spread to many places in the codebase. Not really something I would do for a fringe use case.

**Comment by designermonkey** [comment]
Would there be a way to handle it with the plugin architecture?

**Reply by Seldaek** [reply]
TBH it sounds pretty hard. It may be doable given enough hackery, I am not sure, but I don't see how you'd do this cleanly.
